Source/WebCore/css/CSSProperties.json

```diff
@@ -7497,7 +7497,8 @@
             "codegen-properties": {
                 "conditional-converter": "Perspective",
                 "custom-parser": true,
-                "parser-requires-context-mode": true
+                "parser-requires-context-mode": true,
+                "parser-grammar": ["<length [0,inf]>"]
```

Comment:
Why does this have both `custom-parser` and `parser-grammar`? @weinig Is this even supported?

Reply:
I don’t actually know what this change does, but saw other places that did something similar. I know that the .cpp change alone suffices. I could just remove this.

Reply:
`"parser-grammar": ["<length [0,inf]>"]` actually automatically generates code for parsing. So you probably don't want to use it along with `custom-parser: true`. @weinig, it might be worth adding a warning or build error if someone does this.

Reply:
I'll try removing the `custom-parser` bit and see what happens :)

Reply:
I removed the `CSSProperties.json` changes for now, they weren't material to the fix.

Reply:
The custom parser is still different from the grammar you specified as it supports "none". So if you want them to match it should be:

"parser-grammar": "none | <length [0,inf]>"

I think make it a failure to have both a grammar and custom-parsing makes sense. I'll do that.

Reply:
(Also, don't need the surrounding array: [...] in the grammar property)

Reply:
> I removed the `CSSProperties.json` changes for now, they weren't material to the fix.

I had the browser window open too long. Ignore my comments).